python/ray/data/context.py

```diff
@@ -139,6 +139,22 @@
     os.environ.get("RAY_DATA_OP_RESERVATION_RATIO", "0.5")
 )
 
+# The fraction of the object store capacity that will be used as the default object
+# store memory limit for the streaming executor,
+# when `ReservationOpResourceAllocator` is enabled.
+# deprecate DEFAULT_OBJECT_STORE_MEMORY_LIMIT_FRACTION
+DEFAULT_OBJECT_STORE_MEMORY_LIMIT_FRACTION = float(
+    os.getenv("RAY_DATA_OBJECT_STORE_MEMORY_LIMIT_FRACTION", "0.5")
+)
+# The fraction of the object store capacity that will be used as the default object
+# store memory limit for the streaming executor,
+# when `ReservationOpResourceAllocator` is not enabled.
+DEFAULT_OBJECT_STORE_MEMORY_LIMIT_FRACTION_WO_RESOURCE_RESERVATION = float(
+    os.getenv(
+        "RAY_DATA_OBJECT_STORE_MEMORY_LIMIT_FRACTION_WO_RESOURCE_RESERVATION", "0.25"
+    )
+)
```

Comment:
It makes sense to make the object store memory fraction configurable. 
But ReservationOpResourceAllocator is an implementation detail that I'd like to avoid exposing. 
Maybe it'd be better if we only add one single config item `override_object_store_memory_limit_fraction`. If set, it overrides both cases.

Reply:
Some modifications have been made according to your comment. You can take a look again.